### PR TITLE
Update CI workflow

### PR DESCRIPTION
**Pull request by bjohansebas**
## Changes:

- The setup-node action is used instead of nvm.
- The list-environment task is removed
- The checkout action is updated
- Using Node 20 by default

### CODE REVIEW

**Review by UlisesGascon — APPROVED** [review]

**Review by carlosstenzel — APPROVED** [review]

**Review by jonchurch — APPROVED** [review]

**Comment by jonchurch** [comment]
Heads up that removing the matrix changes the CI job name.

We had `test (16)` required as a job that had to succeed or block merge. I updated the required job in the settings to be `test` after merging this.

That means you will see existing PRs say they're waiting for required job to run, but it won't unless the CI is triggered again via push, close then open of the PR, or a PR approval from someone w/ write. Not a big deal, but that's why you'll see some existing PRs saying they're waiting for required job:

Screenshot 2024-11-16 at 11 01 20 PM

**Pull request merged**
